llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2831,8 +2831,13 @@ SDValue SITargetLowering::LowerFormalArguments(
       RC = &AMDGPU::VGPR_32RegClass;
     else if (AMDGPU::SGPR_32RegClass.contains(Reg))
       RC = &AMDGPU::SGPR_32RegClass;
-    else
-      llvm_unreachable("Unexpected register class in LowerFormalArguments!");
+    else {
+      if (VT == MVT::i1 && Subtarget->isWave64())
```

Comment:
You don't need to make this conditional on wave64, it's essentially checking the same thing twice

Reply:
To operate within the constraints of SILowerI1Copies, we might need to find a way to get this broken into:

```
  %3:sreg_64_xexec = COPY $sgpr0_sgpr1
  %4:vreg_1 = COPY %3
  %5:sreg_64_xexec = COPY %4

```

Reply:
Pls see my comment on SILowerI1Copies.cpp below.